File: src/hasoftware-library/src/hasoftware/api/messages/OutputMessageResponse.java
package hasoftware.api.messages;

import hasoftware.api.FunctionCode;
import hasoftware.api.Message;
import hasoftware.api.classes.OutputMessage;
import hasoftware.cdef.CDEFAction;
import hasoftware.cdef.CDEFMessage;
import hasoftware.cdef.CDEFSystemFlags;
import java.util.LinkedList;
import java.util.List;

public class OutputMessageResponse extends Message {

    private int _action;
    private final List<OutputMessage> _outputMessages = new LinkedList<>();

    public OutputMessageResponse(int transactionNumber) {
        super(FunctionCode.OutputMessage, transactionNumber, CDEFSystemFlags.Response);
        _action = CDEFAction.None;
    }

    public OutputMessageResponse(CDEFMessage cdefMessage) {
        super(cdefMessage);
        _action = cdefMessage.getU8();
        int countOutputMessages = cdefMessage.getU8();
        for (int index = 0; index < countOutputMessages; index++) {
            _outputMessages.add(new OutputMessage(cdefMessage));
        }
    }

    @Override
    public void encode(CDEFMessage cdefMessage) {
        super.encode(cdefMessage);
        cdefMessage.putU8(_action);
        cdefMessage.putU8(_outputMessages.size());
        for (OutputMessage outputMessage : _outputMessages) {
            outputMessage.encode(cdefMessage);
        }
    }

    public int getAction() {
        return _action;
    }

    public void setAction(int action) {
        _action = action;
    }

    public List<OutputMessage> getOutputMessages() {
        return _outputMessages;
    }
}

File: src.net/hasoftware-library/Messages/OutputMessageResponse.cs
using hasoftware.Api;
using hasoftware.Cdef;
using hasoftware.Classes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hasoftware.Messages
{
    public class OutputMessageResponse : Message
    {
        public CdefAction Action { get; private set; }
        public List<OutputMessage> OutputMessages { get; private set; }

        public OutputMessageResponse(int transactionNumber)
            :base(Api.FunctionCode.OutputMessage, transactionNumber, CdefSystemFlags.Response)
        {
            Action = CdefAction.None;
            OutputMessages = new List<OutputMessage>();
        }

        public OutputMessageResponse(CdefMessage cdefMessage)
            :base(cdefMessage)
        {
            Action = (CdefAction)cdefMessage.GetU8();
            int countOutputMessages = cdefMessage.GetU8();
            for (int index = 0; index < countOutputMessages; index++) {
                OutputMessages.Add(new OutputMessage(cdefMessage));
            }
        }

        public override void Encode(CdefMessage cdefMessage) {
            base.Encode(cdefMessage);
            cdefMessage.PutU8((int)Action);
            cdefMessage.PutU8(OutputMessages.Count);
            foreach (var outputMessage in OutputMessages) {
                outputMessage.Encode(cdefMessage);
            }
        }
    }
}
